dspace-api/src/main/java/org/dspace/app/util/RelationshipUtils.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.util;
+
+import java.util.List;
+
+import org.dspace.content.RelationshipType;
+
+public class RelationshipUtils {
+
+    private RelationshipUtils() {
+    }
+
+    /**
+    * Matches two Entity types to a Relationship Type from a set of Relationship Types.
+     *
+    * @param relTypes set of Relationship Types.
+     * @param targetType entity type of target.
+     * @param originType entity type of origin referer.
+     * @return null or matched Relationship Type.
+     */
+    public static RelationshipType matchRelationshipType(List<RelationshipType> relTypes, String targetType,
```

Comment:
Would it be possible to add a Unit Test for this method?  This seems like something we should be able to run through a unit test rather easily.

At a bare minimum, it'd be good to better document the behavior of this method...the current JavaDocs are missing params, and they don't give any examples of what the values of `originType` and `originTypeName` should look like (along with other params).  Examples and/or Unit Tests would make this method easier to understand.

Reply:
Hi Tim, thanks for that. I just pulled this code directly from org.dspace.app.bulkedit.MetadataImport. I created a Util class just to avoid copy/pasting - the intent was that MetadataImport be refactored to refer to this Util method as well.

I can update the javadoc for this method but I don't think I'll have time allocated this month to work on a unit test.